src/cmd-koji-upload

```diff
@@ -203,20 +204,29 @@ class Build(_Build):
         Helper to return if a file should be uploaded
         :param fname: name of file to check against Koji table for uploading.
         :type str
-        :returns bool
+        :returns (bool, str)
 
-        Returns true if the file is known to Koji.
+        The return is tuple is used if the file is found and a message.
         """
         base = os.path.basename(fname)
         check_extension = base.split(".")[-1]
         for extension in COMPRESSION_EXT:
             if fname.endswith(extension):
                 check_extension = base.split(".")[-2]
 
-        found = KOJI_CG_TYPES.get(check_extension)
+        (found, _, arch, _) = KOJI_CG_TYPES.get(
+            check_extension, [None, None, None, None])
+
+        if arch == "arch":
+            farch = os.path.abspath(fname).split('/')[-2]
```

Comment:
I'd write this `os.path.basename(os.path.dirname(fname))` but fine as is too.

More generally, it'd seem cleaner to walk over a build directory and validate the arch name once? 

Anyways not a blocker.

Reply:
I'll test that out. However, I had to use `os.path.abspath` because of rel links in the build directory.